### PR TITLE
*: update etcd-operator and etcd version

### PR DESCRIPTION
**Pull request by hongchaodeng**
This PR is to update image versions of etcd-operator, etcd.
The latest etcd-operator release also fixes scale test, reboot test due to 1.6 upgrade.

### CODE REVIEW

**Comment by pbx0** [comment]
rktbot run disabled tests

**Comment by pbx0** [comment]
rktbot run tests

**Comment by pbx0** [comment]
rktbot run disabled tests

**Comment by pbx0** [comment]
rktbot run disabled tests

**Comment by hongchaodeng** [comment]
rktbot run disabled tests

**Comment by pbx0** [comment]
Looking good. Just need to wait until etcd-operator cuts a release we can use.

**Comment by hongchaodeng** [comment]
rktbot run disabled tests

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by xiang90** [comment]
@hongchaodeng 

can you revert https://github.com/kubernetes-incubator/bootkube/commit/f942073f58e6c46b49ed4942046e01a359792dd6 to see if taint works correctly on multi-node example?

@pbx0 

can you try to enable the taint flag on our testing infra to see if all tests can still pass?

**Comment by hongchaodeng** [comment]
@xiang90 
Tested taint and it (multi-node) worked. Commits pushed here.

**Comment by xiang90** [comment]
@hongchaodeng Thanks!

**Comment by pbx0** [comment]
rktbot run disabled tests

**Comment by Quentin-M** [comment]
Hey.

How is this progressing?
Trying to get https://github.com/coreos/tectonic-installer/pull/135 wrapped up by tomorrow.

**Comment by hongchaodeng** [comment]
@Quentin-M 
We have all the code ready.
We have done the testing last week without taints.
Now we are going through testing with taints. If things go well, we will cut a release in etcd operator and then update this PR.

**Comment by pbx0** [comment]
rktbot run disabled tests

**Comment by pbx0** [comment]
Looking good so far. Both the disabled tests pass on this branch and fail on master. So this fixes the scale test and the reboot test for self-hosted etcd clusters.

**Comment by hongchaodeng** [comment]
etcd-operator v0.2.5 is cut.
Updated PR.

@xiang90 @pbx0 @aaronlevy @squat

**Comment by aaronlevy** [comment]
@hongchaodeng sorry -- didn't see this and just changed vendoring to glide -- can you update this PR with the etcd-operator pinned in the new `glide.yaml`

File: hack/multi-node/user-data.sample [hunk 1]
```diff
@@ -32,6 +32,7 @@ coreos:
           --allow-privileged \
           --hostname-override=${COREOS_PUBLIC_IPV4} \
           --node-labels=node-role.kubernetes.io/master \
+           --register-with-taints=node-role.kubernetes.io/master=:NoSchedule \
```
Comment (aaronlevy) [inline hack/multi-node/user-data.sample]: spacing

File: hack/quickstart/init-master.sh [hunk 2]
```diff
@@ -23,7 +23,7 @@ function configure_etcd() {
         mkdir -p /etc/systemd/system/etcd-member.service.d
         cat << EOF > /etc/systemd/system/etcd-member.service.d/10-etcd-member.conf
 [Service]
-Environment="ETCD_IMAGE_TAG=v3.1.0"
+Environment="ETCD_IMAGE_TAG=v3.1.6"
```
Comment (aaronlevy) [inline hack/quickstart/init-master.sh]: Do we need to pin this version in the vendored client code too?

Comment (hongchaodeng) [inline hack/quickstart/init-master.sh]: Yes. We should pin this in `glide.yaml`.
Bootkube seems to vendor latest etcd and etcd-operator version already. We don't need to update in this PR. I would like to pin in separate PR.

**Comment by hongchaodeng** [comment]
rktbot run disabled tests

File: hack/quickstart/init-master.sh (hunk 2 above)
Comment (aaronlevy) [inline hack/quickstart/init-master.sh]: sgtm

**Pull request merged**